partiql-parser/src/peg.rs

```diff
@@ -33,10 +33,7 @@ impl<'val, R: RuleType> PairsExt<'val, R> for Pairs<'val, R> {
                 }
                 Ok(pair)
             }
-            None => syntax_error(
-                "Expected at one token pair, got nothing!",
-                Position::Unknown,
-            ),
+            None => illegal_state(),
```

Comment:
Did you consider keeping the message so that the error comes with some context?

Reply:
Yeah, the problem statement was a bit unclear to me. My understanding is this:

The None arm is practically impossible to reach, it's an _illegal_ state. Therefore, we could omit the (presumably unhelpful) message and the (unknown) position information and dedicate a unit variant for this state.

Should I keep the message ("Expected at one token pair, got nothing!")?

Reply:
This one is something I probably should've made clearer.  It is something that almost rises to the assertion level, but from a utility perspective could happen if you create a Pest rule that is *silent* and matches will return such an *empty* `Pairs`.

```rust
use pest::iterators::Pairs;
use pest::Parser;
use pest_derive::Parser;

#[derive(Parser)]
#[grammar_inline = "ab = _{ \"ab\" }"]
struct FooParser;

fn main() {
    let pairs: Pairs<Rule> = FooParser::parse(Rule::ab, "ab").unwrap();
    assert_eq!(0, pairs.count());
}
```